src/components/Search.js

```diff
@@ -1,22 +1,44 @@
-import React from 'react';
+import React from "react";
+import Button from "./Button.js";
 
-const Search = props => (
-  <div className="search">
-    <div className="page-header">
-      <h4 className="text-left">Search Bookings</h4>
-    </div>
-    <div className="row search-wrapper">
-      <div className="col">
-        <div className="form-group search-box">
-          <label htmlFor="customerId">Customer Id</label>
-          <div className="search-row">
-            <input type="text" id="customerId" className="form-control" placeholder="Customer Id" />
-            <button className="btn btn-primary fn-submit-name">Search IDs</button>
+export default class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: ""
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
```

Comment:
Because you're listening for updates on the `<input>` and updating the state, this pattern that you're using is called a "controlled component". There's not really a problem with using this pattern, but it is slightly unusual for this particular case. It is often used if we need to transform or keep track of exactly what the user has typed.

React has an alternative pattern (called "uncontrolled component"), were we don't add a onChange handler, and then just get the input when we click the button.

I've just realised that we haven't looked a feature of React that will make this easier in class yet (called *refs*). Hopefully that will make more sense on how we can do that after the class.